[Merged by Bors] - Update Lighthouse book FAQ (#4510)
sigp/lighthouse: book/src/faq.md

This means that the validator client is sending requests to the beacon node. However, as the beacon node is still syncing, it is therefore unable to fulfil the request. The error will disappear once the beacon node is synced. 

### <a name="bn-fork-choice"></a> My beacon node logs `WARN Error signalling fork choice waiter`, what should I do?

An example of the full log is shown below:

```
WARN Error signalling fork choice waiter slot: 6763073, error: ForkChoiceSignalOutOfOrder { current: Slot(6763074), latest: Slot(6763073) }, service: state_advance
```

> This suggests that the computer resources are being overwhelmed. It could be due to high CPU usage or high disk I/O usage. This can happen, e.g., when the beacon node is downloading historical blocks, or when the execution client is syncing. The error will disappear when the resources used return to normal or when the node is synced.

Reviewer: Since this is functionally the same as the `ERRO Aggregate attestation queue full` error, for consistency we could directly copy the text for that section.

Author: > Since this is functionally the same as the `ERRO Aggregate attestation queue full` error, for consistency we could directly copy the text for that section.

Do you mean having the same content of answers for these two questions?

Reviewer: Yep that's right. I think we may as well use the same answer for both

Author: Revised as suggested, thank you

### <a name="bn-queue-full"></a> My beacon node logs `ERRO Aggregate attestation queue full`, what should I do?

An example of the full log is shown below:
```
ERRO Aggregate attestation queue full, queue_len: 4096, msg: the system has insufficient resources for load, module: network::beacon_processor:1542
```

> This indicates that the system resources are overloaded. The above error will usually be seen when the node is doing some resource-intensive processes, e.g., the execution client is syncing, or the beacon node is downloading historical blocks. You can check the CPU, RAM or disk usage to see if they are being overloaded. Once the resource-intensive processes completed, the error should be gone.

Reviewer: Just a small wording suggestion here.

```suggestion
This indicates that the system resources are overloaded. The above error will usually be seen when the node is doing some resource-intensive processes, e.g., the execution client is syncing, or the beacon node is downloading historical blocks. You can check the CPU, RAM or disk usage to confirm they are being overloaded. The error should resolve once the resource-intensive processes have completed.
```

- the internet is working well
- you have sufficient peers

You can see more information on the [Ethstaker KB](https://ethstaker.gitbook.io/ethstaker-knowledge-base/help/missed-attestations). 

> Another cause for missing attestations is due to the slowness in processing the block. When this happens, the debug log will show (the debug log can be found under `$datadir/beacon/logs`):

Reviewer: ```suggestion
Another cause for missing attestations is delays during block processing. When this happens, the debug logs will show (debug logs can be found under `$datadir/beacon/logs`):
```

### <a name="vc-head-vote"></a> Sometimes I miss the attestation head vote, resulting in penalty. Is this normal?

In general, it is unavoidable to have some penalties occasionally. This is particularly the case when you are assigned to attest on the first slot of an epoch and if the proposer of that slot releases the block late, then you will get penalised for missing the target and head votes. Your attestation performance does not only depend on your own setup, but also on everyone elses performance.

You could also check for the sync aggregate participation percentage on block explorers such as [beaconcha.in](https://beaconcha.in/). A low sync aggregate participation percentage (e.g., 60-70%) indicates that the block that you are assigned to attest to may be published late. As a result, your validator fails to correctly attest to the block. 

> Another possible reason for missing the head vote is due to reorg. The reorg happens due to the block is published late, and the proposer at block `n+1` reorged blocks `n` from the canonical chain. Due to the reorg, block `n` was never included in the chain.  If you are assigned to attest at slot `n`, the head reward will be missed.

Reviewer: ```suggestion
Another possible reason for missing the head vote is due to a chain "reorg". A reorg can happen if the proposer publishes block `n` late, and the proposer of block `n+1` builds upon block `n-1` instead of `n`. This is called a "reorg". Due to the reorg, block `n` was never included in the chain.  If you are assigned to attest at slot `n`, it is possible you may still attest to block `n` despite most of the network recognizing the block as being late. In this case you will miss the head reward.
```

### <a name="net-port"></a> How to change the TCP/UDP port 9000 that Lighthouse listens on?
Use the flag ```--port <PORT>``` in the beacon node. This flag can be useful when you are running two beacon nodes at the same time. You can leave one beacon node as the default port 9000, and configure the second beacon node to listen on, e.g., ```--port 9001```.

### <a name="net-subnet"></a> Lighthouse `v4.3.0` introduces a change where a node will subscribe to only 2 subnets in total. I am worried that this will impact my validators return.

> Previously, having more validators means subscribing to more subnets. Since the change, a node will only subscribe to 2 subnets in total. Hence, this will bring about significant reductions in bandwidth for nodes with multiple validators. 

Reviewer: ```suggestion
Previously, having more validators means subscribing to more subnets. Since the change, a node will now only subscribe to 2 subnets in total. This will bring about significant reductions in bandwidth for nodes with multiple validators. 
```

> While having subscribed to more subnets, or all 64 subnets could ensure having peers on more or all subnets, these subscriptions are consuming resources and bandwidth. This does not significantly increase the performance of the node, it does help out other nodes on the network. 

Reviewer: ```suggestion
While subscribing to more subnets can ensure you have peers on a wider range of subnets, these subscriptions consume resources and bandwidth. This does not significantly increase the performance of the node, however it does benefit other nodes on the network. 
```

> Therefore, there is no significant gain for a node to subscribe to more subnets. However, if you would still like to subscribe to all subnets, you can use the flag `subscribe-all-subnets`. This could improve the block rewards by 1-5%, though it comes at the cost of a much higher bandwidth requirement.

Reviewer: ```suggestion
If you would still like to subscribe to all subnets, you can use the flag `subscribe-all-subnets`. This may improve the block rewards by 1-5%, though it comes at the cost of a much higher bandwidth requirement.
```
